Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -2003,19 +2042,32 @@ namespace Babylon
 
                     current.insert(inputSource.ID);
 
-                    auto found = m_idToInputSource.find(inputSource.ID);
-                    if (found == m_idToInputSource.end())
+                    auto inputSourceFound = m_idToInputSource.find(inputSource.ID);
+                    if (inputSourceFound == m_idToInputSource.end())
                     {
                         // Create the new input source, which will have the correct spaces associated with it.
                         m_idToInputSource.insert({inputSource.ID, CreateXRInputSource(inputSource, env)});
+                        
+                        //Now that input Source is created, create a gamepad object for the input source
+                        inputSourceFound = m_idToInputSource.find(inputSource.ID);
```

Comment:
Since this adds entries alongside m_idToInputSource, it should also remove entries when m_idToInputSource removes them (line 2120).